src/inference/src/os/lin/lin_system_conf.cpp

```diff
@@ -22,36 +24,169 @@ struct CPU {
+
+                            _cores++;
+                            _proc_type_table[ALL_PROC]++;
+                            _proc_type_table[EFFICIENT_CORE_PROC]++;
+                        }
+                        n_group++;
+
+                    } else {
+                        core_1 = std::stoi(cache_1_info);
+
+                        _cpu_mapping_table[core_1][CPU_MAP_CORE_ID] = _cores;
+                        _cpu_mapping_table[core_1][CPU_MAP_CORE_TYPE] = MAIN_CORE_PROC;
+                        _cpu_mapping_table[core_1][CPU_MAP_GROUP_ID] = n_group;
+
+                        _cores++;
+                        n_group++;
+                        _proc_type_table[ALL_PROC]++;
+                        _proc_type_table[MAIN_CORE_PROC]++;
+                    }
+                }
+                return;
+            };
+
+            for (int n = 0; n < _processors; n++) {
+                if (-1 == _cpu_mapping_table[n][CPU_MAP_SOCKET_ID]) {
+                    std::ifstream cache_3_file("/sys/devices/system/cpu/cpu" + std::to_string(n) +
+                                               "/cache/index3/shared_cpu_list");
+                    std::string cache_3_info;
+                    std::getline(cache_3_file, cache_3_info);
+                    std::string::size_type pos = 0;
+                    std::string::size_type endpos = 0;
+                    std::string sub_str;
+                    int core_1;
+                    int core_2;
+
+                    while (1) {
+                        if ((endpos = cache_3_info.find('-', pos)) != std::string::npos) {
+                            sub_str = cache_3_info.substr(pos, endpos);
+                            core_1 = std::stoi(sub_str);
+                            sub_str = cache_3_info.substr(endpos + 1);
+                            core_2 = std::stoi(sub_str);
+
+                            for (int m = core_1; m <= core_2; m++) {
+                                _cpu_mapping_table[m][CPU_MAP_SOCKET_ID] = _sockets;
+                                updateProcMapping(m);
+                            }
+                        } else if (pos != std::string::npos) {
+                            sub_str = cache_3_info.substr(pos);
+                            core_1 = std::stoi(sub_str);
+                            _cpu_mapping_table[core_1][CPU_MAP_SOCKET_ID] = _sockets;
+                            updateProcMapping(core_1);
+                            endpos = pos;
+                        }
+
+                        if ((pos = cache_3_info.find(',', endpos)) != std::string::npos) {
+                            pos++;
+                        } else {
+                            break;
+                        }
+                    }
+                    _sockets++;
+                }
             }
-            if (0 == key.find("physical id")) {
-                socketId = std::stoi(value);
+        } else {
```

Comment:
As we agreed offline: lets create follow-up task on mapping of old "cpuinfo" path into new CPU table represntation. It should simplify StreamExecutor implementation logic.
The task can be done after another refactoring activities.